apps/kg/models/general_models.py

```diff
@@ -462,3 +462,36 @@ def finish_async_update(self):
         """Terminate the async update for entity embedding.
         """
         self.entity_emb.finish_async_update()
+
+
+    def pull_model(self, client, pos_g, neg_g):
+        with th.no_grad():
+            entity_id = F.cat(seq=[pos_g.ndata['id'], neg_g.ndata['id']], dim=0)
+            relation_id = pos_g.edata['id']
+            entity_id = F.tensor(np.unique(F.asnumpy(entity_id)))
```

Comment:
I suggest we can implement F.unique by ourself.
torch has th.unique which will be efficient that converting it to numpy and back

Reply:
I find that the numpy operation here is not a bottleneck compared with the network. So I think we can just use numpy for simplify.